src/api/artifact/controller.go

```diff
@@ -32,17 +36,19 @@ type Controller interface {
 	// creates it if it doesn't exist. If tags are provided, ensure they exist
 	// and are attached to the artifact. If the tags don't exist, create them first
 	Ensure(ctx context.Context, repository *models.RepoRecord, digest string, tags ...string) (err error)
-	// List artifacts according to the query and option
+	// List artifacts according to the query, specify the properties returned with option
 	List(ctx context.Context, query *q.Query, option *Option) (total int64, artifacts []*Artifact, err error)
 	// Get the artifact specified by ID
 	Get(ctx context.Context, id int64, option *Option) (artifact *Artifact, err error)
 	// Delete the artifact specified by ID
 	Delete(ctx context.Context, id int64) (err error)
-	// DeleteTag deletes the tag specified by ID
-	DeleteTag(ctx context.Context, id int64) (err error)
-	// UpdatePullTime updates the pull time for the artifact. If the tag is provides, update the pull
+	// Tags returns the tags according to the query, specify the properties returned with option
+	Tags(ctx context.Context, query *q.Query, option *TagOption) (total int64, tags []*Tag, err error)
+	// DeleteTag deletes the tag specified by tagID
+	DeleteTag(ctx context.Context, tagID int64) (err error)
+	// UpdatePullTime updates the pull time for the artifact. If the tagID is provides, update the pull
 	// time of the tag as well
-	UpdatePullTime(ctx context.Context, artifactID int64, tag string, time time.Time) (err error)
+	UpdatePullTime(ctx context.Context, artifactID int64, tagID int64, time time.Time) (err error)
```

Comment:
What if the tagID is invalid or not exist while artifact ID does exist?

Reply:
Will report a NOT_FOUND_ERROR